Extension/src/LanguageServer/extension.ts

```diff
@@ -28,6 +29,8 @@ let realActivationOccurred: boolean = false;
 let tempCommands: vscode.Disposable[] = [];
 let activatedPreviously: PersistentWorkspaceState<boolean>;
 
+export let customConfigurationProviders: CustomConfigurationProvider[] = [];
```

Comment:
nit: I don't really like to export global vars for every file in the extension to use if we can avoid it. Maybe we should just move all of this (array and registration method) into protocolFilter.ts?  Or else, export a "provideConfiguration" function and pass a function with your logic from protocolFilter to it.